packages/esm-patient-chart-app/src/visit/visit-form/visit-form.component.tsx

```diff
@@ -55,18 +55,40 @@ import { useForm, Controller, FormProvider } from 'react-hook-form';
 import { z } from 'zod';
 import { zodResolver } from '@hookform/resolvers/zod';
 
-const visitFormSchema = z.object({
-  visitDate: z.date(),
-  visitTime: z.string(),
-  timeFormat: z.enum(['PM', 'AM']),
-  programType: z.string().optional(),
-  visitType: z.string().refine((value) => value != '', 'Visit type is required'),
-  visitLocation: z.object({
-    display: z.string(),
-    uuid: z.string(),
-  }),
-  visitAttributes: z.record(z.string()).optional(),
-});
+const visitAttributeTypeSchema = z.record(z.object({ required: z.boolean() }));
+
+const visitFormSchema = z
```

Comment:
We can modify the schema based on the config. since the config is the only thing which knows both the uuid and if the field is required or not.